### ISSUE TITLE
Cahaba Codebase Refactor and Cleanup

### ISSUE THREAD

**Issue by nickchadwick-noaa** [NOAA-OWP/inundation-mapping#422]
This Issue is to track the overall progress of cleaning up the Cahaba Repo. The following line items are to be made into separate issues with their own deeper descriptions and discussions. The goal of this list is to clean out any unused code and files, unify the formatting and languages of all the remaining code, and to organize the repository for future contributions.

**Won't Impact fim_4:**
- [x] Remove `Dockerfile.prod`, rename `Dockerfile.dev` to just `Dockerfile`, and remove `.dockerignore`. #432
- [x] Clean up `Dockerfile` and remove any unused* packages or variables. #432
- [x] Remove any unused* Python packages from the `Pipfile`. #432
- [x] Move the `CHANGELOG.md`, `SECURITY.md`, and `TERMS.md` files to the `/docs` folder. #432
- [x] Remove any unused* scripts in the `/tools` and `/src` folders. #432
- [x] Move `tools/preprocess` scripts into `tools/` #432
- [x] Ensure all scripts in the `/src` folder have their code in functions and are being called via a `__main__` function (This will help with implementing memory profiling fully). #432
- [x] Make memory profiling run as an option flag only.
- [ ] Standardize code #194 
- [ ] Validate argument inputs #197 
- [ ] Develop unit tests for code #195 
- [x] Remove unused* imports from Python scripts #52
- [ ] Validation, logging of `fim_run.sh` HUC success / failure
- [ ] Address bug where summary log file writes "Exit status = 0" for HUCs with non-zero exit statuses.

*unused by both Cahaba and fim_4.

**Might Impact fim_4**
- [ ] Convert the `run_by_unit.sh` script from Bash to Python.
- [ ] Convert the `fim_run.sh` script from Bash to Python.
- [ ] Finish implementing `memory_profiling` in the `fim_run`, `run_by_unit`, and `/src/*` scripts.

**Comment by BradfordBates-NOAA** [comment]
@nickchadwick-noaa Can you please note for each item whether or not it might harm the fim_4 forked repo? Let's make two lists, one called "Won't Impact fim_4" and another list called "Might Impact fim_4". For example, "Converting the `run_by_unit.sh` script from Bash to Python might impact fim_4". Again, this is just a best guess. We'll coordinate with DS later to confirm.

**Comment by BradfordBates-NOAA** [comment]
@fernando-aristizabal Thanks for checking in on this. After July 4 we are planning to hold a session with the DS team to iron out a refactor/clean-up plan for fim_4 sensitive items. Until then, we are planning to knock out some of the harmless stuff. If you see something on the list, like the Dockerfile refactoring plans, that cause concern, please let us know.

**Comment by nickchadwick-noaa** [comment]
@fernando-aristizabal I will of course post any scripts we plan to remove, to ensure that they are not being used by anyone.

Initially, I had created a separate `Dockerfile.prod` to be used in a production environment. This Dockerfile is very similar to the Dev, except that it includes the codebase directly in the image instead of using a docker volume to link the codebase. We ended up not using this production Docker image when we moved to a "production" environment. To my knowledge this `Dockerfile.prod` is not used (unless you guys are using it),  so I was just going to remove it.
